chest/core.py

```diff
@@ -196,7 +196,12 @@ def __setitem__(self, key, value):
 
     def __del__(self):
         if self._explicitly_given_path:
-            self.flush()
+            if os.path.exists(self.path):
+                self.flush()
+            else:
```

Comment:
This else should mate with the first `if`.  `__del__` is called on garbage collection.  We only want to delete the chest on garbage collection if the user didn't specify a path when creating the chest and we used a temporary path instead.